### ISSUE TITLE
importing pre-trained weights for multiband image training

### ISSUE THREAD

**Issue by onurbarut** [thtrieu/darkflow#507]
Hey all,

I am trying to train YOLOv2 (or its tiny versions) to detect only one class in multiband (actually four band, with NearIR) images. I constructed the dataset according to VOC example. 

My dataset contains about 1200 images with 500x500 size. Objects I try to detect are about 50x50 to 150x150 pixels and they could be found around 3-5 up to 20 or 25 in a single image. I changed the cfg file filters accordingly.

My first question is that, can I import the pre-trained weights for (for example) tiny-yolo? Since my data is 500x500x4, the first convolution kernels' size are mismatched. Is there any suggestions to import the first conv. kernels' 3x3x3x16 and concat with the 3rd dimension with randomly initialized weight to match the first kernel 3x3x4x16?

Since I have trouble with the steps above, I am trying to train my 1 class dataset from scratch, and I haven't succeeded yet. The loss is around 5 after 480k iterations (took about two days on GTX1060 6GB, batch size is 4) and I can't lower it even though I reduced the learning rate down to 1e-9. The results on the test image is not so promising, as the predicted bounding boxes are 2 or 3 time larger than the ground truth. Could changing my anchor box dimensions be helpful? Could the problem be that my objects are too small to detect? Should my dataset include images that do not contain any object?

One last question: what are the parameters of the last layer stands for? for 5 anchor boxes and 1 class, it is like 5*(1+5), first 5 is number of anchor boxes, 1 stands for the number of classes, and 5 stands for x, y, w, h, and Confidence? What is that confidence? Is it that the given cell contains an (any) object? Given that the cell contains an object, since I have only one class, do I have to add that "1" term? I am confused here. Could anyone help me get the relationship of the last volume (13x13x30) and the loss calculation?

Thanks.
